### PR TITLE
Clarified IAM docs section + added policy example

### LINKED ISSUE #3059: S3 Permissions Docs
**What happened**:

Customer has reported a lot of critical information in cloud-trail logs, around S3 access to viewing recorded sessions. 

We need to `ListObjectVersions`. `s3:ListBucketVersions` & `s3:GetObjectVersion` 

**What you expected to happen**:

We need to provider clearer information for s3 permissions, in Docs, Terraform and CloudFormation examples.

### PR DESCRIPTION
**Pull request by aelkugia**
Resolves: #3059

### CODE REVIEW

File: docs/4.1/aws_oss_guide.md [hunk 1]
```diff
@@ -124,6 +124,63 @@ proxy and provides the TCP connections needed for Teleport proxy SSH connections
...
+     }
+   ]
+ }
+```
+
+#### IAM for DynamoDB
+
+In order to grant an IAM user access to DynamoDB make sure that the IAM role assigned to Teleport is configured with proper permissions. 
+
+An example policy is shown below:
```
Comment (benarent) [inline docs/4.1/aws_oss_guide.md]: I would be inclined to add a not that `u-west-1:123456789012:table/prod.teleport.auth` should be updated with the customers dynamoDB instance.

File: docs/4.1/aws_oss_guide.md [hunk 2]
```diff
@@ -124,6 +124,63 @@ proxy and provides the TCP connections needed for Teleport proxy SSH connections
 IAM is the recommended tool for creating service access. This guide will follow the 
 best practice of principle of least privilege (PoLP). 
 
+#### IAM for Amazon S3
+
+In order to grant an IAM user in your AWS account access to one of your buckets, `example.s3.bucket` you will need to grant the following permissions: `s3:ListBucket`, `s3:ListBucketVersions`, `s3:PutObject`, `s3:GetObject`, `s3:GetObjectVersion`
+
+An example policy is shown below:
```
Comment (benarent) [inline docs/4.1/aws_oss_guide.md]: I would also note that `example.s3.bucket` should be replaced by a real bucket.

**Review by benarent — APPROVED** [review]
Looks good, I might be inclined to add

**Comment by aelkugia** [comment]
Made the updates mentioned regarding notes with clarification, let me know what you think on comment left on #3059 regarding redundancy

**Review by klizhentas — APPROVED** [review]

**Review by benarent — APPROVED** [review]

**Pull request merged**